index.js

```diff
@@ -144,29 +185,30 @@ module.exports = {
               root,
               previous: [root], // TODO calculate previous for members tangle
             },
-            // likely incorrect group tangle and this will be overwritten by publish()
-            // we just add it here to make the spec pass
+            // likely incorrect group tangle and this will be overwritten by
+            // publish(), we just add it here to make the spec pass
             group: {
               root,
               previous: [root],
             },
           },
-          recps,
+          recps: [groupId, ...feedIds],
         }
 
         if (opts.text) content.text = opts.text
 
-        if (!addMemberSpec(content))
-          return cb(new Error(addMemberSpec.errorsString))
+        // TODO: this should accept bendybutt-v1 feed IDs
+        // if (!addMemberSpec(content))
+        //   return cb(new Error(addMemberSpec.errorsString))
 
-        publish(content, (err, msg) => {
+        findOrCreateInvitationsFeed((err, invitationsFeed) => {
           if (err) return cb(err)
 
-          //TODO: this is an optimization, use the db query instead for now
-          //keystore.group.registerAuthors(groupId, feedIds, (err) => {
-          //  if (err) return cb(err)
-          return cb(null, msg)
-          //})
+          addGroupTangle(content, (err, content) => {
+            if (err) return cb(err)
+
+            prunePublish(ssb, content, invitationsFeed.keys, cb)
```

Comment:
can we call this publishAndPrune?

Reply:
wait, I don't understand that function. It seems to loop, so it will do many creates? I thought it would prune down the tangle and then publish.

Reply:
Then forget all of this :)

Reply:
Done